File: very_good_flutter_plugin/__brick__/{{#ios}}{{project_name.snakeCase()}}_ios{{/ios}}/ios/{{project_name.snakeCase()}}_ios/Sources/{{project_name.snakeCase()}}_ios/{{project_name.pascalCase()}}Plugin.swift
import Flutter

public class {{project_name.pascalCase()}}Plugin: NSObject, FlutterPlugin, {{project_name.pascalCase()}}Api {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let binaryMessenger = registrar.messenger()
    let instance = {{project_name.pascalCase()}}Plugin()
    {{project_name.pascalCase()}}ApiSetup.setUp(binaryMessenger: binaryMessenger, api: instance)
    registrar.publish(instance)
  }

  func getPlatformName(completion: @escaping (Result<String?, Error>) -> Void) {
    completion(.success("iOS"))
  }
}

File: very_good_flutter_plugin/__brick__/{{#android}}{{project_name.snakeCase()}}_android{{/android}}/android/src/main/kotlin/{{org_name.pathCase()}}/{{project_name.pascalCase()}}Plugin.kt
package {{org_name.dotCase()}}

import {{project_name.pascalCase()}}Api
import io.flutter.embedding.engine.plugins.FlutterPlugin

class {{project_name.pascalCase()}}Plugin : FlutterPlugin, {{project_name.pascalCase()}}Api {
    companion object {
        private const val TAG = "{{project_name.pascalCase()}}Plugin"
    }

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        {{project_name.pascalCase()}}Api.setUp(binding.binaryMessenger, this)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        {{project_name.pascalCase()}}Api.setUp(binding.binaryMessenger, null)
    }

    override fun getPlatformName(callback: (Result<String?>) -> Unit) {
        callback(Result.success("Android ${android.os.Build.VERSION.RELEASE}"))
    }
}